File: CS-Examples/19_PivotTables/RepeatItemLabels.cs
using Spire.Xls;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Xls.Core.Spreadsheet.PivotTables;

namespace RepeatItemLabels
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            // Create a workbook
            Workbook workbook = new Workbook();

            // Load an excel file including pivot table
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\RepeatItemLabelsExample.xlsx");

            // Get the first worksheet
            Worksheet sheet = workbook.Worksheets[0];

            // Add an empty worksheet 
            Worksheet sheet2 = workbook.CreateEmptySheet();
            sheet2.Name = "Pivot Table";

            // Define the data range for the pivot table
            CellRange dataRange = sheet.Range["A1:D9"];

            // Create a pivot cache using the data range
            PivotCache cache = workbook.PivotCaches.Add(dataRange);

            // Add a pivot table to the pivot sheet using the pivot cache
            PivotTable pt = sheet2.PivotTables.Add("Pivot Table", sheet.Range["A1"], cache);

            // Set the VendorNo field as a row field and specify its header caption
            var r1 = pt.PivotFields["VendorNo"];
            r1.Axis = AxisTypes.Row;
            pt.Options.RowHeaderCaption = "VendorNo";
            r1.Subtotals = SubtotalTypes.None;

            // Enable repeating item labels for the VendorNo field
            r1.RepeatItemLabels = true;

            // Enable repeating item labels for the OnHand field
            pt.PivotFields["OnHand"].RepeatItemLabels = true;

            // Set the row layout type to tabular
            pt.Options.RowLayout = PivotTableLayoutType.Tabular;

            // Set the Desc field as an additional row field
            var r2 = pt.PivotFields["Desc"];
            r2.Axis = AxisTypes.Row;

            // Add the OnHand field as a data field with the label "Sum of onHand"
            pt.DataFields.Add(pt.PivotFields["OnHand"], "Sum of onHand", SubtotalTypes.None);

            // Set the built-in style for the pivot table appearance
            pt.BuiltInStyle = PivotBuiltInStyles.PivotStyleMedium12;

            // Specify the filename for the resulting workbook
            String result = "RepeatItemLabels_result.xlsx";

            // Save the modified workbook to a file
            workbook.SaveToFile(result, ExcelVersion.Version2010);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            // Launch the document
            FileViewer(result);
        }

        private void FileViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }

        private void Form1_Load(object sender, EventArgs e)
        {

        }
    }
}

File: VB-Examples/19_PivotTables/RepeatItemLabels.vb
Imports Spire.Xls
Imports System.ComponentModel
Imports System.Text
Imports Spire.Xls.Core.Spreadsheet.PivotTables

Namespace RepeatItemLabels
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object
            Dim workbook As New Workbook()

            ' Load an Excel file from the specified path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\RepeatItemLabelsExample.xlsx")

            ' Get the first worksheet in the workbook
            Dim sheet As Worksheet = workbook.Worksheets(0)

            ' Create a new empty worksheet
            Dim sheet2 As Worksheet = workbook.CreateEmptySheet()
            sheet2.Name = "Pivot Table"

            ' Define the range of data to be used for the PivotTable
            Dim dataRange As CellRange = sheet.Range("A1:D9")

            ' Create a PivotCache based on the data range
            Dim cache As PivotCache = workbook.PivotCaches.Add(dataRange)

            ' Add a PivotTable to sheet2 with the specified name, source range, and cache
            Dim pt As PivotTable = sheet2.PivotTables.Add("Pivot Table", sheet.Range("A1"), cache)

            ' Configure the first field ("VendorNo") in the PivotTable
            Dim r1 = pt.PivotFields("VendorNo")
            r1.Axis = AxisTypes.Row
            pt.Options.RowHeaderCaption = "VendorNo"
            r1.Subtotals = SubtotalTypes.None
            r1.RepeatItemLabels = True

            ' Configure the second field ("OnHand") in the PivotTable
            pt.PivotFields("OnHand").RepeatItemLabels = True

            ' Set the row layout of the PivotTable to Tabular
            pt.Options.RowLayout = PivotTableLayoutType.Tabular

            ' Configure the third field ("Desc") in the PivotTable
            Dim r2 = pt.PivotFields("Desc")
            r2.Axis = AxisTypes.Row

            ' Add a data field ("OnHand") to the PivotTable
            pt.DataFields.Add(pt.PivotFields("OnHand"), "Sum of onHand", SubtotalTypes.None)

            ' Set the built-in style of the PivotTable to PivotStyleMedium12
            pt.BuiltInStyle = PivotBuiltInStyles.PivotStyleMedium12

            ' Specify the output file name
            Dim result As String = "RepeatItemLabels_result.xlsx"

            ' Save the modified workbook to the specified file path in Excel 2010 format
            workbook.SaveToFile(result, ExcelVersion.Version2010)

            ' Release the resources used by the workbook
            workbook.Dispose()

            'View the document
            FileViewer(result)
		End Sub

		Private Sub FileViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load

		End Sub
	End Class
End Namespace
